array.go

```diff
@@ -913,3 +913,59 @@ func countElement(array []interface{}, element interface{}) int {
 	}
 	return count
 }
+
+// Transform runs the passed function on all the Elements in the array.
+//
```

Comment:
Let's also mention that it return a new array, and original one is not affected.